ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/momentumBasedController/optimization/DynamicsMatrixCalculator.java

```diff
@@ -2,99 +2,149 @@
 
 import org.ejml.data.DMatrixRMaj;
 import org.ejml.dense.row.CommonOps_DDRM;
-
 import us.ihmc.commonWalkingControlModules.controllerCore.WholeBodyControlCoreToolbox;
-import us.ihmc.matrixlib.MatrixTools;
 import us.ihmc.mecano.algorithms.CompositeRigidBodyMassMatrixCalculator;
-import us.ihmc.mecano.multiBodySystem.RevoluteJoint;
 import us.ihmc.mecano.multiBodySystem.interfaces.FloatingJointBasics;
-import us.ihmc.mecano.multiBodySystem.interfaces.JointBasics;
-import us.ihmc.mecano.multiBodySystem.interfaces.OneDoFJointBasics;
 import us.ihmc.mecano.multiBodySystem.interfaces.RigidBodyBasics;
-import us.ihmc.mecano.spatial.SpatialForce;
 import us.ihmc.mecano.spatial.interfaces.WrenchReadOnly;
-import us.ihmc.mecano.tools.MultiBodySystemTools;
+import us.ihmc.robotics.MatrixMissingTools;
 import us.ihmc.robotics.screwTheory.FloatingBaseRigidBodyDynamicsCalculator;
 import us.ihmc.robotics.screwTheory.GravityCoriolisExternalWrenchMatrixCalculator;
 
+/**
+ * Helper class for computing the rigid body dynamics matrices. In general, the rigid body dynamics are represented by
+ * <p>
+ *    H(q) qDdot + C(q, qDot) + J<sup>T</sup>(q) &rho; = [0; &tau;<sup>T</sup>]<sup>T</sup>
+ * </p>
+ * <p>
+ *    They can be further partition into the floating base component, demarcated by (.)<sub>base</sub>, and the actuated component, demarcated by
+ *    (.)<sub>body</sub>. These correspond to the top rows in the above equation for the base and the bottom rows for the actuated portion.
+ * </p>>
+ * <p>
+ *    In inverse dynamics classes, the joint accelerations (qDdot) and contact forces (&rho;) are typically solved for, and then the joint torques (&tau;) are
+ *    computed. This class is designed to provide the affine terms to compute these values.
+ * </p>
+ */
 public class DynamicsMatrixCalculator
```

Comment:
This class was a travesty and makes me embarrassed that I wrote it. There's some weird stuff at the end, still, that's only used in the tests.